packages/graphql-hooks/src/LocalGraphQLClient.js

```diff
@@ -0,0 +1,54 @@
+import GraphQLClient from './GraphQLClient'
+
+/** Local version of the GraphQLClient which only returns specified queries
+ * Meant to be used as a way to easily mock and test queries during development. This client never contacts any actual server.
+ * Queries are given in the form of an object of functions.
+ * Example:
+ ```
+    const localQueries = {
+      [allPostsQuery]: () => ({
+        allPosts: [
+          {
+            id: 1,
+            title: 'Test',
+            url: 'https://example.com'
+          }
+        ]
+      }),
+      [createPostMutation]: () => ({ createPost: { id: 1 } }),
+    }
+    const client = new LocalGraphQLClient({ localQueries })
+  ```
+ */
+class LocalGraphQLClient extends GraphQLClient {
+  constructor(config = {}) {
+    super(config)
+    this.localQueries = config.localQueries
+    if (!this.localQueries) {
+      throw new Error(
+        'LocalGraphQLClient: `localQueries` object required in the constructor options'
+      )
+    }
+  }
+
+  verifyConfig() {
+    // Skips all config verification from the parent class because we're mocking the client
+  }
+
+  request(operation) {
+    if (this.localQueries[operation.query]) {
+      return Promise.resolve(
+        this.localQueries[operation.query](
+          operation.variables,
+          operation.operationName
+        )
+      ).then(data => ({ data }))
```

Comment:
Good point, I'll look at adding that

Reply:
As I was implementing this, I realised the normal GraphQL client never actually throws errors, it just returns an `error` object (which is not itself an error), as done by this code:

```
  generateResult({ fetchError, httpError, graphQLErrors, data }) {
    const errorFound = !!(
      (graphQLErrors && graphQLErrors.length > 0) ||
      fetchError ||
      httpError
    )
    return !errorFound
      ? { data }
      : { data, error: { fetchError, httpError, graphQLErrors } }
  }
```

Which  means, the local mock should not throw either but instead should return the "error" as part of the object the query returns.

But to keep the convenience of being able to write mock functions that just return data directly ( ie. `() => ({ createPost: { id: 1 } })` instead of `() => ({ data: { createPost: { id: 1 } }})`, we could use a `LocalGraphQLError` class as a way to then check in the `LocalGraphQLClient` if the object returned is an instance of that class (then assign it to the `error` object) or just data (then assign it to `data`

Reply:
Sorry, I've missed some bits in my explaination, but I'm happy that you filled the gaps and reached the same conclusions :)